Filters/src/DetectorStepFilter_module.cc

```diff
@@ -31,25 +38,30 @@ namespace mu2e {
 
-	fhicl::Atom<double> minTrkStepEnergy { Name("MinimumTrkStepEnergy"), Comment("Minimum Trk step energy"), -std::numeric_limits<double>::max() };
-	fhicl::Atom<double> minCaloStepEnergy { Name("MinimumCaloStepEnergy"), Comment("Minimum Calo step energy"), -std::numeric_limits<double>::max() };
-	fhicl::Atom<double> minCrvStepEnergy { Name("MinimumCrvStepEnergy"), Comment("Minimum Crv step energy"), -std::numeric_limits<double>::max() };
+	fhicl::Atom<double> minTrkStepEnergy { Name("MinimumTrkStepEnergy"), Comment("Minimum Trk step energy"), 0.0 };
+	fhicl::Atom<double> minCaloStepEnergy { Name("MinimumCaloStepEnergy"), Comment("Minimum Calo step energy"), 0.0 };
+	fhicl::Atom<double> minCrvStepEnergy { Name("MinimumCrvStepEnergy"), Comment("Minimum Crv step energy"), 0.0 };
 
-	fhicl::Atom<double> minPartMom { Name("MinimumPartMom"), Comment("Minimum particle momentum"), -std::numeric_limits<double>::max() };
-	fhicl::Atom<double> maxPartMom { Name("MaximumPartMom"), Comment("Maximum particle momentum"), std::numeric_limits<double>::max() };
+	fhicl::Atom<double> minPartMom { Name("MinimumPartMom"), Comment("Minimum particle momentum"), 0.0 };
+	fhicl::Atom<double> maxPartMom { Name("MaximumPartMom"), Comment("Maximum particle momentum"), maxE_ };
 
 	fhicl::Atom<bool> orRequirements { Name("ORRequirements"), Comment("Take the logical OR of requirements, otherwise take the AND"), true };
 
 	fhicl::Atom<unsigned> minNTrkSteps { Name("MinimumTrkSteps"), Comment("Minimum number of good tracker steps"), 10};
 	fhicl::Atom<double> minSumCaloStepE { Name("MinimumSumCaloStepE"), Comment("Minimum E sum of good calorimeter steps (MeV) "), 50.0 };
 	fhicl::Atom<unsigned> minNCrvSteps { Name("MinimumCrvSteps"), Comment("Minimum number of good CRV steps"), 3 };
 
-	fhicl::Atom<unsigned> maxNTrkSteps { Name("MaximumTrkSteps"), Comment("Maximum number of good tracker steps"), 1000000};
-	fhicl::Atom<double> maxSumCaloStepE { Name("MaximumSumCaloStepE"), Comment("Maximum E sum of good calorimeter steps (MeV) "), 1.0e8 };
-	fhicl::Atom<unsigned> maxNCrvSteps { Name("MaximumCrvSteps"), Comment("Maximum number of good CRV steps"), 1000000 };
+	fhicl::Atom<unsigned> maxNTrkSteps { Name("MaximumTrkSteps"), Comment("Maximum number of good tracker steps"), maxN_ };
+	fhicl::Atom<double> maxSumCaloStepE { Name("MaximumSumCaloStepE"), Comment("Maximum E sum of good calorimeter steps (MeV) "),  maxE_ };
+	fhicl::Atom<unsigned> maxNCrvSteps { Name("MaximumCrvSteps"), Comment("Maximum number of good CRV steps"), maxN_ };
 
 	fhicl::Sequence<int> keepPDG { Name("KeepPDG"), Comment("PDG particle codes to keep") };
 
+	fhicl::Atom<bool> timeCut { Name("TimeCut"), Comment("Make time for good step"), false};
+	fhicl::OptionalAtom<double> maxTime { Name("MaximumTime"), Comment("Maximum time for good step (ns since POT)")};
+	fhicl::OptionalAtom<double> minTime { Name("MinimumTime"), Comment("Minimum time for good step (ns since POT)")};
```

Comment:
Having 3 time cuts is confusing. What is "make time"?  Please expand these Comments to make clear what  the cuts do and why we need 3.

Reply:
On the second look, the first atom is bool, not a cut.
There is a better pattern to use here.   Please define a substructure with min and max time value,
and in place of the bool use an OptionalTable<substructure>.  For example see
VolumeInfoMixerConfig in EventMixing/inc/Mu2eProductMixer.hh

Reply:
If this config were used more than once, or if art let me reuse the config struct in the code I would agree, but since they will be turned into individual module variables anyways it seems more work than is warranted.
I can make the comments clearer.